utils/misc.py

```diff
@@ -0,0 +1,10 @@
+def ordinal(n: int) -> str:
+    match n%10:
+        case 1:
+            return f'{n}st'
+        case 2:
+            return f'{n}nd'
+        case 3:
+            return f'{n}rd'
+        case _:
+            return f'{n}th'
```

Comment:
This will have issues withe 11, 12 and 13. A possible change could be something like
```python
def ordinal(n: int) -> str:
    if n < 0:
        # Throw error
    def suffix(n:int) -> str:
        if n in [11, 12, 13]: return 'th'
        if n%10 == 1: return 'st'
        if n%10 == 2: return 'nd'
        if n%10 == 3: return 'rd'
        return 'th'
    return f'{n}{suffix(n)}`
```

Reply:
This is probably not the best, you could use the match again, or remove the sub-function entirely, up to you.

Reply:
I kept the match statements because I have been excited to use them.😄 
I edited the implementation such that inputs like "111" or "2012" also return with the correct suffix.
```py
def ordinal(n: int) -> str:
    if n < 0:
        raise ValueError('Negative ordinal - ordinal must be passed a non-negative integer.')

    match n//10 % 10:  # for numbers trailing with 11, 12, or 13
        case 1:
            return f'{n}th'
    match n%10:
        case 1:
            return f'{n}st'
        case 2:
            return f'{n}nd'
        case 3:
            return f'{n}rd'
        case _:
            return f'{n}th'
```